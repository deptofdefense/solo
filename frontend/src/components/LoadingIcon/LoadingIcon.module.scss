.error {
  color: #b51d09;
}
