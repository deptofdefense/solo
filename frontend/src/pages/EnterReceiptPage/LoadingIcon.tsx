import React from "react";
import classNames from "classnames";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faSpinner,
  faExclamationCircle,
  faCheck
} from "@fortawesome/free-solid-svg-icons";
import { LoadingStatus } from "solo-types";
import classes from "./EnterReceiptPage.module.scss";

const LoadingIcon: React.FC<LoadingStatus> = ({ loading, error }) => (
  <span>
    <FontAwesomeIcon
      spin={loading}
      icon={loading ? faSpinner : error ? faExclamationCircle : faCheck}
      className={classNames({
        [classes.error]: error
      })}
    />
  </span>
);

export default LoadingIcon;
