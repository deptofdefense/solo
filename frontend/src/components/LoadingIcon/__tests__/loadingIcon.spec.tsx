import React from "react";
import { render } from "test-utils";
import { LoadingIcon } from "../LoadingIcon";

describe("LoadingIcon component", () => {
  it("renders loading icon when loading", async () => {
    const { asFragment } = render(<LoadingIcon loading error={false} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders error icon on fetch error", async () => {
    const { asFragment } = render(
      <LoadingIcon loading={false} error={true} message="some error" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders check icon on fetch success", async () => {
    const { asFragment } = render(
      <LoadingIcon loading={false} error={false} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
