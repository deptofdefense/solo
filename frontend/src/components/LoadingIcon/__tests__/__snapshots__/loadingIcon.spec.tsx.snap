// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIcon component renders check icon on fetch success 1`] = `
<DocumentFragment>
  <span>
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-check fa-w-16 "
      data-icon="check"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
        fill="currentColor"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`LoadingIcon component renders error icon on fetch error 1`] = `
<DocumentFragment>
  <span>
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-exclamation-circle fa-w-16 error"
      data-icon="exclamation-circle"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
        fill="currentColor"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`LoadingIcon component renders loading icon when loading 1`] = `
<DocumentFragment>
  <span>
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-spinner fa-w-16 fa-spin "
      data-icon="spinner"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
        fill="currentColor"
      />
    </svg>
  </span>
</DocumentFragment>
`;
